Makefile

```diff
@@ -117,7 +117,16 @@ copy-core-for-cli:
 # If you need other binaries, look at the build-xxx rules below.
 .PHONY: minimal-build
 minimal-build:
-	dune build _build/install/default/bin/semgrep-core
+	$(eval $@_TMP := $(shell mktemp -t dune-output))
+	@# Save the output of dune so we can provide more helpful error messages in
+	@# some cases
+	2>&1 dune build _build/install/default/bin/semgrep-core | tee $($@_TMP)
+	@grep -q "Error during linking" $($@_TMP) && ( \
```

Comment:
for this PR can you generate one of the error messages and confirm it's readable and easy to debug?

Reply:
I saw one in pad's comments. Would be nice if it was top level in the PR comment for future reference. Also would be nice to maybe add a comment here or in flags.sh explaining how to read it.

Reply:
I mean, it will be as readable as any linker error. I don't think an comment explaining how to read a linker error here would be that useful. What do you think would be unclear about the linking issue itself? I agree with some of the other comments about more instruction on how to change/add to flags.sh.